test/core/networking/websockets/session_test.dart

```diff
@@ -0,0 +1,408 @@
+import 'dart:async';
+import 'dart:io';
+import 'package:test/test.dart';
+
+import 'package:flyde/core/networking/websockets/session.dart';
+import 'package:flyde/core/networking/websockets/middleware.dart';
+
+import '../../../helpers/value_hook.dart';
+import '../../../helpers/wait_for.dart';
+
+//? The WebSocket uses in this file are not race conditions,
+//? because of how the dart eventloop is implemented.
+//? The internal WebSocket connect is !scheduled! and executed when the main strain waits (await)
```

Comment:
Please don't explain why your implementation does not crash instantly multiple times, everywhere and highlighted.
If it is important only in the tests, add it at the specific locations. Otherwise add this to the appropriate docs

Reply:
It is added at the exact locations where this might be confusing to people who read the code for the first time.

Reply:
So you can remove the extra note at the top. This is written three times in this file. That's too much

Reply:
Removed the header comment